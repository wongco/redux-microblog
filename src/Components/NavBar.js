import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

const StyledNav = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  background-color: lightgray;
  width: 100%;
  border-radius: 0.5em;
  box-shadow: 0px 0px 18px gray;
  padding: 3em;
`;

const StyledNavTitle = styled.p`
  font-size: 300%;
  margin: 0px;
`;

const StyledNavLinkContainer = styled.div`
  display: flex;
  box-sizing: border-box;
  flex-direction: row;
`;

const StyledNavLink = styled(NavLink)`
  padding-right: 10px;
  text-decoration: none;
  color: blue;
`;

class NavBar extends Component {
  render() {
    return (
      <StyledNav>
        <div>
          <StyledNavTitle>Microblog</StyledNavTitle>
          <p>Get in the Rithm of Blogging!</p>
        </div>
        <StyledNavLinkContainer>
          <StyledNavLink to="/">Blog</StyledNavLink>
          <StyledNavLink to="/new">Add a new post</StyledNavLink>
        </StyledNavLinkContainer>
      </StyledNav>
    );
  }
}

export default NavBar;
