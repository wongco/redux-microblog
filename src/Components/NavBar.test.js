import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from './NavBar';

it('renders without crashing', () => {
  shallow(<NavBar />);
});

it('matches snapshot for default criteria', () => {
  const wrapper = shallow(<NavBar />);
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
