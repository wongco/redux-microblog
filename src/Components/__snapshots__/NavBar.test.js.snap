// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for default criteria 1`] = `
<styled.div>
  <div>
    <styled.p>
      Microblog
    </styled.p>
    <p>
      Get in the Rithm of Blogging!
    </p>
  </div>
  <styled.div>
    <Styled(NavLink)
      to="/"
    >
      Blog
    </Styled(NavLink)>
    <Styled(NavLink)
      to="/new"
    >
      Add a new post
    </Styled(NavLink)>
  </styled.div>
</styled.div>
`;
